src/frontends/pytorch/src/utils_quantize.cpp

```diff
@@ -168,6 +169,57 @@ std::shared_ptr<QuantizedPtNode> cast_quantized_fw_node(std::shared_ptr<Node> no
     return quant_node;
 }
 
+std::shared_ptr<Node> u4_compression_stack(const NodeContext* context,
+                                            const std::deque<ov::Output<ov::Node>>& list_elems,
+                                            int64_t axis) {
+    // Part 1: Detect pattern
+
+    if (list_elems.size() != 2)
+        return nullptr;
+    auto bitwise_and = cast_fw_node(list_elems[0].get_node_shared_ptr(), "aten::bitwise_and");
+    if (!bitwise_and)
+        return nullptr;
+    auto bitwise_shift = cast_fw_node(list_elems[1].get_node_shared_ptr(), "aten::bitwise_right_shift");
+    if (!bitwise_shift)
+        return nullptr;
+
+    // TODO: Use context (if not nullptr) to mark newly created nodes
```

Comment:
The result of this compression is a single u4 Constant, it can be marked outside of this function, `context` is not required to be passed.